### PR TITLE
Systemd - disable privatedevices as it may not work for some devices

### PR DESCRIPTION
**Pull request by magikstm**
### 1. What does this change do, exactly?
Sets privatedevices in systemd to 'false' as it may not work for some devices.

### 2. Please link to the relevant issues.
I didn't create one yet. It was discussed in the forum recently here:
https://caddy.community/t/systemd-fails-to-load-caddy-on-a-raspberry-pi/3170

It is related to: 
https://github.com/mholt/caddy/issues/1853
https://serverfault.com/questions/871635/error-running-otherwise-working-caddy-under-systemd-v232
https://caddy.community/t/error-running-otherwise-working-caddy-under-systemd-v232/2638

More details on this feature here:
https://www.freedesktop.org/software/systemd/man/systemd.exec.html#PrivateDevices=

### 3. Which documentation changes (if any) need to be made because of this PR?
None.

### 4. Checklist

- [ ] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I am willing to help maintain this change if there are issues with it later

### CODE REVIEW

**Comment by francislavoie** [comment]
Maybe add a note to the [README](https://github.com/mholt/caddy/blob/master/dist/init/linux-systemd/README.md) with a similar note to your comment, plus a mention that it can be turned on for some other devices for added security?

@mholt this is the first I've seen PrivateDevices, so sorry, can't really give much better of a review 😛

**Comment by mholt** [comment]
Ha, alright, thanks @francislavoie  :smile:

File: dist/init/linux-systemd/caddy.service [hunk 1]
```diff
@@ -30,8 +30,8 @@ LimitNPROC=512
 
 ; Use private /tmp and /var/tmp, which are discarded after caddy stops.
 PrivateTmp=true
-; Use a minimal /dev
-PrivateDevices=true
+; Use a minimal /dev (May bring additional security if switched to 'true', but it may not work on Raspberry Pi's or other devices, so it has been disabled in this dist.)
+PrivateDevices=false
```
Comment (mholt) [inline dist/init/linux-systemd/caddy.service]: What if this line (and its associated comment) was just removed altogether?

Comment (magikstm) [reply dist/init/linux-systemd/caddy.service]: I'm unsure all devices set it to "false" by default. This line forces it.

As it adds security, I think it'll make it's possible use easier if left in that file.

**Pull request merged**